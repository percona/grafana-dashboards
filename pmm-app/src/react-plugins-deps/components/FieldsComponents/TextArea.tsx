import React from 'react';
import { useField } from 'react-final-form-hooks';

interface TextAreaFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form: any;
}
export const TextAreaField = ({ name, prefix, placeholder, label, required, form }: TextAreaFieldInterface) => {
  const field = useField(name, form);

  return <textarea {...field.input} rows={5} className="input-field input-field--textarea input-field--dark" placeholder={placeholder}></textarea>;
};
