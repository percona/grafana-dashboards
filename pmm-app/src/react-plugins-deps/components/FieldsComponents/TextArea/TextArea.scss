#antd {
  .text-area-field-wrapper {
    display: flex;
    flex-direction: column;
    align-content: flex-start;
  }

  .text-area-field-wrapper .error-message {
    color: indianred;
  }
}
