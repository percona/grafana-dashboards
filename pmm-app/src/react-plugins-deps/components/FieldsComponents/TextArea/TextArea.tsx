import React from 'react';
// import { useField } from 'react-final-form';
import { useField } from 'react-final-form-hooks';

interface TextAreaFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  form?: any;
  style?: any;
}
export const TextAreaField = ({ name, prefix, placeholder, label, required, form, style }: TextAreaFieldInterface) => {
  const field = useField(name, form);
  return (
    <textarea
      {...field.input}
      rows={5}
      className="input-field input-field--textarea input-field--dark"
      style={style}
      placeholder={placeholder}
    ></textarea>
  );
};
