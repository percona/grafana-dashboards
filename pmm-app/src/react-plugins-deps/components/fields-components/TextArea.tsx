import React, {FunctionComponent} from "react";
import { Field } from "react-final-form";

const TextAreaComponent = ({
  input,
  meta,
  placeholder = "",
  prefix,
  label,
  required,
  labelCol,
  wrapperCol,
  ...rest
}) => {
  return (
    <textarea
      onChange={input.onChange}
      rows={5}
      className="input-field input-field--textarea input-field--dark"
      placeholder={placeholder}
    ></textarea>
  );
};

interface TextAreaFieldInterface {
  required?: boolean;
  name: string;
  "data-cy"?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
}
export const TextAreaField = ({
  name,
  prefix,
  placeholder,
  label,
  required,
  ...rest
}: TextAreaFieldInterface) => {
  return (
    <Field
      {...rest}
      prefix={prefix}
      name={name}
      component={TextAreaComponent as FunctionComponent}
      placeholder={placeholder}
      label={label}
      required={required}
    />
  );
};
