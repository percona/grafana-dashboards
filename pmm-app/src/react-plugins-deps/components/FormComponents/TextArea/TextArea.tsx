import React from 'react';
import { useField } from 'react-final-form';
import './TextArea.scss';
import cx from 'classnames';
interface TextAreaFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  style?: any;
  className?: any;
}
export const TextAreaField = ({ name, placeholder, style, className }: TextAreaFieldInterface) => {
  const { input, meta } = useField(name);
  return (
    <div className="text-area-field-wrapper">
      <textarea
        {...input}
        rows={5}
        className={cx('input-field input-field--textarea input-field--dark', className)}
        style={style}
        placeholder={placeholder}
      />
      {meta.error && meta.touched && <span className="error-message">{meta.error}</span>}
    </div>
  );
};
