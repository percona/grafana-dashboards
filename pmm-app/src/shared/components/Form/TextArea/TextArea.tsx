import React from 'react';
import { useField } from 'react-final-form';
import { cx } from 'emotion';
import './TextArea.scss';

interface TextAreaFieldInterface {
  required?: boolean;
  name: string;
  dataQa?: string;
  label?: string;
  prefix?: string;
  placeholder?: string;
  style?: any;
  className?: any;
}
export const TextAreaField = ({
  name, placeholder, style, className, dataQa,
}: TextAreaFieldInterface) => {
  const { input, meta } = useField(name);

  return (
    <div className="text-area-field-wrapper">
      <textarea
        {...input}
        rows={5}
        className={cx('input-field input-field--textarea input-field--dark', className)}
        style={style}
        placeholder={placeholder}
        data-qa={dataQa}
      />
      {meta.error && meta.touched && <span className="error-message">{meta.error}</span>}
    </div>
  );
};
