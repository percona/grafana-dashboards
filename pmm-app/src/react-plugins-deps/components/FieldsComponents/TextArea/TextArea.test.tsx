// Just a stub test
import { TextAreaField } from './TextArea';
import React from 'react';
import renderer from 'react-test-renderer';

describe('TextArea field test', () => {
  it('TextArea renders correct without props', () => {
    const component = renderer.create(<TextAreaField />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
