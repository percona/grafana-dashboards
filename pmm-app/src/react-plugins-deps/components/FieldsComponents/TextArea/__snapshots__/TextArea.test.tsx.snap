// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea field test TextArea renders correct without props 1`] = `
<textarea
  className="input-field input-field--textarea input-field--dark"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  rows={5}
  value=""
/>
`;
